src/Arbitrary.hs

```diff
@@ -8,14 +8,14 @@
 
 {-# OPTIONS_GHC -fno-warn-orphans -Werror #-}
 
-module Arbitrary (topLevelDomains, loremIpsum) where
+module Arbitrary (topLevelDomains, loremIpsum, generate, arbitrary) where
```

Comment:
I am not that fond of re-exports.  They seem convenient early-on, but they increase overall complexity of the import graph, and make reading the code more challenging (unless we stick with explicit import lists all over).

I don't want to change anything here and now, just a general remark.